core/interfaces/rebase.py

```diff
@@ -284,6 +296,95 @@ def base_ref(self):
 
         return base_ref
 
+    def nearest_branch(self, branch, default="master"):
```

Comment:
Do you mean into it's own helper class? Initially I had this in `BranchesMixin` since it's basically a branch detection mechanism, then I moved it here since felt it's very much related to rebase interface.

But, perhaps it's better if we move this into it's own mixin module: `git_mixins.rebase` to potentially be used by other (rebase) commands. How do you feel about this?